bundles/org.openhab.ui/web/src/pages/settings/rules/script/blockly-editor.vue

```diff
@@ -328,156 +266,388 @@
+            </value>
+          </block>
+          <block type="oh_playmedia_sink_volume">
+            <value name="sinkName">
+              <shadow type="oh_audiosink_dropdown" />
+            </value>
+            <value name="volume">
+              <shadow type="audioSlider">
+                <field name="volume">5</field>
+              </shadow>
+            </value>
+          </block>
+          <block type="oh_playstream_sink">
+            <value name="sinkName">
+              <shadow type="oh_audiosink_dropdown" />
+            </value>
+          </block>
+          <block type="oh_stopstream_sink">
+            <value name="sinkName">
+              <shadow type="oh_audiosink_dropdown" />
+            </value>
+          </block>
+          <block type="oh_say">
+            <value name="textToSay">
+              <shadow type="text">
+                <field name="TEXT">text to say</field>
+              </shadow>
+            </value>
+            <value name="deviceSink">
+              <shadow type="oh_audiosink_dropdown" />
+            </value>
+            <value name="voice">
+              <shadow type="oh_voices_dropdown" />
+            </value>
+          </block>
+        </category>
+        <category name="Logging/Output">
+          <block type="oh_log">
+            <value name="message">
+              <shadow type="text">
+                <field name="TEXT">abc</field>
+              </shadow>
+            </value>
+          </block>
+          <block type="oh_print">
+            <value name="message">
+              <shadow type="text">
+                <field name="TEXT">abc</field>
+              </shadow>
+            </value>
+          </block>
+        </category>
       </category>
-
       <sep />
-      <category name="Variables" colour="%{BKY_VARIABLES_HUE}" custom="VARIABLE" />
-      <category name="Functions" colour="%{BKY_PROCEDURES_HUE}" custom="PROCEDURE" />
+      <category
+        name="Variables"
+        colour="%{BKY_VARIABLES_HUE}"
+        custom="VARIABLE" />
+      <category
+        name="Functions"
+        colour="%{BKY_PROCEDURES_HUE}"
+        custom="PROCEDURE" />
     </xml>
   </div>
 </template>
 
 <style lang="stylus">
-.blockly-editor
-  display block
-  top calc(100%)
-  height calc(100%)
-  width 100%
-  .blockly
-    height 100%
-    .blocklyMainBackground
-      stroke inherit
-.blocklyDropDownDiv
-  z-index 9000
+.blockly-editor {
+  display: block;
+  top: calc(100%);
+  height: calc(100%);
+  width: 100%;
+
+  .blockly {
+    height: 100%;
+
+    .blocklyMainBackground {
+      stroke: inherit;
+    }
+  }
+}
+
+.blocklyDropDownDiv {
+  z-index: 9000;
+}
 </style>
 
 <script>
 import Blockly from 'blockly'
+import '@blockly/block-plus-minus'
+
 import Vue from 'vue'
 
 import defineOHBlocks from '@/assets/definitions/blockly/ohblocks'
+import defineOHBlocksTimers from '@/assets/definitions/blockly/ohblocks_timers'
```

Comment:
Maybe these imports should be moved to an index.js file in `@/assets/definitions/blockly/index.js`, with a function to define all blocks in one go.

Reply:
I wouldn't mind but as you are the expert on this I would recommend you rather do after we are done with all the other changes, so I don't break anything?